File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        for i in range(len(flowerbed)):
            if n == 0:
                return True
            # use variables to make it readable
            is_empty = (flowerbed[i] == 0)
            left_empty = (i == 0 or flowerbed[i - 1] != 1)
            right_empty = (i == len(flowerbed) - 1 or flowerbed[i + 1] != 1)
            if is_empty and left_empty and right_empty:
                n -= 1
                flowerbed[i] = 1
        
        return n == 0
        

File: 0605-can-place-flowers/0605-can-place-flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
    let i = 0;

    while (i < flowerbed.length) {
        if (!n) return true;
        if (flowerbed[i] === 0) {
            if ((i == 0 || flowerbed[i - 1] === 0) && (i === flowerbed.length - 1 || flowerbed[i + 1] === 0) ) {
                i++;
                n--;
                }
        } 
        i++;
        
    }

    console.log(n)

    return n === 0;
};